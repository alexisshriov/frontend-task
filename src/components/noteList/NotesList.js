import React from 'react';
import NodeRow from '../noteRow/NoteRow';

class NotesList extends React.Component {
    state = {
        filteredNotes: this.props.notes,
        filterText: ''

    }

    static getDerivedStateFromProps(props, state) {
        const filteredNotes = props.notes.filter(note => (
            note.title.toUpperCase().includes(state.filterText.toUpperCase()) || note.text.toUpperCase().includes(state.filterText.toUpperCase())
        ))
        console.log('excuted')
        return { filteredNotes: filteredNotes }
    }

    handleChange = (event) => {
        const value = event.target.value
        this.setState({ filterText: value })
    }

    render() {
        return (
            <div style={{ width: 300 }}>
                <div style={{ margin: 7 }} >
                    <label>Filter:</label><input type='text' onChange={this.handleChange}></input>
                </div>
                {this.state.filteredNotes.map(note => <NodeRow key={note.id} note={note} deleteNote={this.props.deleteNote} selectNote={this.props.selectNote} />)}
            </div>
        );
    }
}

export default NotesList;

