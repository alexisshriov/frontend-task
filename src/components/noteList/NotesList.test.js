import React from 'react';
import { shallow } from 'enzyme';
import NotesList from './NotesList';

describe('NoteList', () => {
  it('should render NoteList correctly', () => {
    const notes = [{id: 'd39gr', title: 'note title test1', text: 'note text test1'}, {id: 'ds35', title: 'note title test2', text: 'note text test2'}]
    const component = shallow(<NotesList notes={notes} />);
    expect(component).toMatchSnapshot();
  });
});