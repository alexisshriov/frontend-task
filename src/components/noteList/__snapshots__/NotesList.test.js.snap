// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList should render NoteList correctly 1`] = `
<div
  style={
    Object {
      "width": 300,
    }
  }
>
  <div
    style={
      Object {
        "margin": 7,
      }
    }
  >
    <label>
      Filter:
    </label>
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
  <NoteRow
    key="d39gr"
    note={
      Object {
        "id": "d39gr",
        "text": "note text test1",
        "title": "note title test1",
      }
    }
  />
  <NoteRow
    key="ds35"
    note={
      Object {
        "id": "ds35",
        "text": "note text test2",
        "title": "note title test2",
      }
    }
  />
</div>
`;
